component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/clustering/kmeans/KMeansIncrementalSPExtension.java

```diff
@@ -0,0 +1,305 @@
+import org.wso2.siddhi.query.api.definition.AbstractDefinition;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * performs kmeans with incremental update
+ */
+@Extension(
+        name = "kMeansIncremental",
+        namespace = "streamingml",
+        description = "Performs K-Means clustering on a streaming data set. Data points can be of any dimension and " +
+                "the dimensionality is calculated from number of parameters. " +
+                "All data points to be processed by an instance of class Clusterer should be of the same " +
+                "dimensionality. The Euclidean distance is taken as the distance metric. " +
+                "The algorithm resembles Sequential K-Means Clustering at " +
+                "https://www.cs.princeton.edu/courses/archive/fall08/cos436/Duda/C/sk_means.htm ",
+        parameters = {
+                @Parameter(
+                        name = "model.name",
+                        description = "The name for the model that is going to be created/reused for prediction",
+                        type = {DataType.STRING}
+                ),
+                @Parameter(
+                        name = "decay.rate",
+                        description = "this is the decay rate of old data compared to new data. " +
+                                "Value of this will be in [0,1]. 0 means only old data used and" +
+                                "1 will mean that only new data is used",
+                        optional = true,
+                        type = {DataType.FLOAT},
+                        defaultValue = "0.01f"
+                ),
+                @Parameter(
+                        name = "number.of.clusters",
+                        description = "The assumed number of natural clusters (numberOfClusters) in the data set.",
+                        type = {DataType.INT}
+                ),
+                @Parameter(
+                        name = "coordinate.values",
+                        description = "This is a variable length argument. Depending on the dimensionality of data " +
+                                "points we will receive coordinates along each axis.",
+                        type = {DataType.DOUBLE} //how to give multiple types?
+                )
+
+        },
+        returnAttributes = {
+                @ReturnAttribute(
+                        name = "euclideanDistanceToClosestCentroid",
+                        description = "Represents the Euclidean distance between the current data point and the " +
+                                "closest centroid.",
+                        type = {DataType.DOUBLE}
+                ),
+                @ReturnAttribute(
+                        name = "closestCentroidCoordinatei",
```

Comment:
extra 'i'

Reply:
Oh.. so i is the index. Better to use closestCentroidCoordinate as the name and then explain the behaviour in the description.

Reply:
done